connectors/acoustic/main.py

```diff
@@ -0,0 +1,198 @@
+        If you specify the delete node, you must also specify the schema node.
+
+    - schema: (optional) specifies primary key columns for each entity.
+        you must be very consistent with the schema node and
+        the primary key columns to avoid any unwanted behavior.
+        If you don’t specify the schema, Fivetran appends the data.
+
+    - hasMore: is an indicator for Fivetran to make a follow-up call for fetching the next set of data.
+        Fivetran keeps making repeated calls until it receives hasMore = false.
+
+    More details: https://fivetran.com/docs/functions#responseformat
+    """
+
+    state: Dict[str, Any] = attr.ib(validator=attr.validators.instance_of(dict))
+    schema: Dict[Any, Any] = attr.ib(validator=attr.validators.instance_of(dict))
+    inserts: Dict[Any, Any] = attr.ib(
+        validator=attr.validators.instance_of(dict), default=dict()
+    )
+    deletes: Dict[Any, Any] = attr.ib(
+        validator=attr.validators.instance_of(dict), default=dict()
+    )
+    hasMore: bool = attr.ib(validator=attr.validators.instance_of(bool), default=False)
+
+
+SCHEMA = {
+    "agg_metric_org": {
+        "primary_key": ["mailing_id", "report_id"],
+    },
+    "reporting_raw_recipient_data_export": {
+        "primary_key": [
+            "recipient_id",
+            "mailing_id",
+            "campaign_id",
+            "customer_id",
+        ],  # TODO: figure out primary_key: Recipient Id,Mailing Id,Report Id,Campaign Id,Content Id,CustomerID?
+    },
+}
+
+REPORTING_RAW_RECIPIENT_DATA_EXPORT_REQUEST_BODY_FILE = (
+    "reporting_raw_recipient_data_export.xml.jinja"
+)
+
+
+def main(request: Dict[Any, Any]) -> dict:
+    """
+    Function that will be executed by Fivetran.
+
+    The standard format of `request` is a JSON object with the following fields:
+        `agent`: informal object
+        `state`: contains bookmark that marks the data Fivetran has already synced
+        `secrets`: optional JSON object that contains access keys or API keys, other config
+
+    Returns FivetranResponse as a dictionary.
+    """
+
+    date_format = "%m/%d/%Y %H:%M:%S"
+    date_delta = timedelta(days=1)  # actual frequency of fetching data
+    # Fivetran frequency is just responsible for trigerring the job
+
+    default_start_date = request["secrets"]["default_start_date"]  # type: datetime
+    state_start_date = datetime.strptime(
+        request["state"].get("next_date_start"), date_format
+    )  # type: datetime
+    date_start = state_start_date or default_start_date  # type: datetime
+
+    current_system_date = datetime.now()
+
+    # Making sure that date_start is 1 day older than current_date
+    if (current_system_date - date_start) < date_delta:
+        logging.info(
+            "Skipping run as the difference between current_system_date (%s) \
+             - date_start(%s) is less than date_delta (%s)"
+            % (current_system_date, date_start, date_delta)
+        )
```

Comment:
You might be able to do push some of this to Airflow rather than in the connector if you think that's a good idea